src/neps/utils/common.py

```diff
@@ -231,6 +232,17 @@ def get_searcher_data(searcher: str, searcher_path: Path | str | None = None) ->
     return data
 
 
+def get_value(obj: Any):
+    if isinstance(obj, (str, int, float, bool, type(None))):
+        return obj
+    elif isinstance(obj, dict):
+        return {key: get_value(value) for key, value in obj.items()}
+    elif isinstance(obj, list):
+        return [get_value(item) for item in obj]
+    else:
+        return obj.__name__
```

Comment:
What's the point of this function in the larger context? I saw it above just to do printing. Seems to just convert most object values to their name, otherwise their raw value if `str(), int(), float(), bool() or None`?

If the goal is just to have strings or data that is serializable, might be better to rename it as `get_value` in `neps.util.commons` doesn't really speak to its use and will likely not get re-used by anyone else, as they won't know why or where to use it.

Reply:
Some arguments are passed as instances or classes in neps.run like this example [here](https://github.com/automl/neps/blob/master/neps_examples/experimental/hierarchical_architecture_hierarchical_GP.py#L110-L140). When serializing it to the yaml, an error would occur.
Likewise when changing it to strings, something like <neps.optimizers.bayesian_optimization.optimizer.BayesianOptimization object at 0xHEX...> would then be saved. So I opted for a cleaner look by only saving the class name instead. 

This can probably be better if we allow the users to pass strings instead of classes and then we map those strings internally.